CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentResultsHtmlFileExport.cs

```diff
@@ -0,0 +1,141 @@
+﻿using CharlieBackend.Business.Helpers;
+using CharlieBackend.Core.DTO.Dashboard;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Business.Services.FileServices.ExportFileServices.Html
+{
+    class StudentResultsHtmlFileExport : FileExport<StudentsResultsDto>
+    {
+        #region private constants
+
+        private const int CourseColumnNumber = 0;
+        private const int StudentGroupColumnNumber = 1;
+        private const int ThirdColumnNumber = 2;
+        private const int FourthColumn = 3;
+
+        #endregion
+
+        private void FillFile(StudentsResultsDto data)
+        {
+            if (data == null)
+            {
+                return;
+            }
+
+            var table = new StringBuilder();
+
+            bool averageVisitsExistAndValid = data.AverageStudentVisits != null && data.AverageStudentVisits.Any();
```

Comment:
please add list initialization in dto and remove null check

Reply:
Done!